### DISCUSSION TITLE
Custom Membership Fields not displaying on membership form

### DISCUSSION THREAD

**Discussion by Sp0run** [liedekef/events-made-easy#670] [General]
I have a family membership form which uses two #_FAMILYMEMBERS to pull a secondary form for additional members. I have a custom yes/no drop down field which is a required field for each member. I have tried setting the field to a members type field and it will display correctly on the parent form but display as #_FIELD{36} on the additional members form.

I have also tried setting the field to a memberships type field but it does not display on the parent membership form and on the additional members form it displays as "#_FIELD{36} again.

Am I doing something wrong or is this a buglet?

**Comment by liedekef** [comment]
You can't have multiple #_FAMILYMEMBERS present. Only 1 time is needed and you should ask for the number of family members via #_FAMILYCOUNT (so EME knows how many times the form per family member should be repeated). The next version of EME will prevent the multiple usage and enforce both to be present if not present.
For the rest it should be a people-type custom field.

**Comment by Sp0run** [comment]
Yes, sorry I should've said it uses both #_FAMILYCOUNT & #_FAMILYMEMBERS as per the below excerpt. I have multiple memberships on my site and the field that I have added is pertinent to the specific membership. If I change this to a people field it will be global for that person rather than an attribute of that members membership. If there's no other way, it'll have to go as a people field. 

> <h3>Additional Family Members</h3>
> Up to 2 adults and 3 children
> 
> Additional Family Member Count: #_FAMILYCOUNT
> 
> #_FAMILYMEMBERS

As always, thanks for your support.

**Reply by liedekef** [reply]
Currently the logic is that the membership is "shared" (identical) between the master membership account and the family members, I'll need to test if it works ok with fields of type "member" too.
If you want to test this on your end (I'm currently not at home), go into eme_formfields.php and change line 3545 (in the function eme_replace_membership_familyformfields_placeholders) from:
`			if ( ! empty( $formfield ) && $formfield['field_purpose'] == 'people' ) {`
to:
`            if ( ! empty( $formfield ) && in_array( $formfield['field_purpose'], [ 'generic', 'members', 'people' ] ) ) {`

**Comment by Sp0run** [comment]
Thanks Franky, I'll have a play when I finish work..

**Reply by liedekef** [reply]
I tried (and verified code): currenly only people-type fields will work there.

**Reply by liedekef** [reply]
Your question forced me to take a good look at the family-code in detail, so I changed the code and the next version will allow fields of type "member" too.

**Reply by Sp0run** [reply]
Oh, thats really good of you, thanks!

**Comment by Sp0run** [comment]
I think we've introduced a bug. On my membership form if #_FAMILYCOUNT is set to zero, it still pulls the form for additional family members

**Reply by liedekef** [reply]
are you testing out the dev version?

**Reply by Sp0run** [reply]
I'm using version 2.5.18

**Reply by liedekef** [reply]
Then you should open this as a bug next time. But it is a bug, this fixes it: https://github.com/liedekef/events-made-easy/commit/4eaf3c75ccf1a79f93958031ddf0d636a0149f6c

**Reply by Sp0run** [reply]
My apologies, but sweet as, thanks

**Reply by liedekef** [reply]
btw, feel free to test the dev-version if you want (for the membership thingie)

**Reply by Sp0run** [reply]
Yeah sure, I’ll have a look when I can 👍
